lib/ProductOpener/Images.pm

```diff
@@ -1309,28 +1309,35 @@ sub process_image_crop ($user_id, $product_id, $id, $imgid, $angle, $normalize,
 		$h = $z;
 	}
 
-	print STDERR
-		"image_crop.pl - imgid: $imgid - crop_size: $crop_size - x1: $x1, y1: $y1, x2: $x2, y2: $y2, w: $w, h: $h\n";
-	$log->trace("calculating geometry",
-		{crop_size => $crop_size, x1 => $x1, y1 => $y1, x2 => $x2, y2 => $y2, w => $w, h => $h})
-		if $log->is_trace();
-
-	my $ox1 = int($x1 * $ow / $w);
-	my $oy1 = int($y1 * $oh / $h);
-	my $ox2 = int($x2 * $ow / $w);
-	my $oy2 = int($y2 * $oh / $h);
-
-	my $nw = $ox2 - $ox1;    # new width
-	my $nh = $oy2 - $oy1;
-
-	my $geometry = "${nw}x${nh}\+${ox1}\+${oy1}";
-	$log->debug("geometry calculated",
-		{geometry => $geometry, ox1 => $ox1, oy1 => $oy1, ox2 => $ox2, oy2 => $oy2, w => $w, h => $h})
-		if $log->is_debug();
-	if ($nw > 0) {    # image not cropped
-		my $imagemagick_error = $source->Crop(geometry => $geometry);
-		($imagemagick_error)
-			and $log->error("could not crop to geometry", {geometry => $geometry, error => $imagemagick_error});
+	# avoid a potential divide by zero error
+	if (!$w or !$h) {
```

Comment:
That will avoid the division by zero, but I'm not sure what happens later on.
It seems the problem was that $new_product_ref->{images}{$imgid} is not defined.
So in this case, we would not crop the image, but select it.

Reply:
And so what would you advise to do.

Reply:
It's fine as is I guess. Or we could add a check earlier: if the imgid is not in images, then we don't select and crop it. I don't really understand the conditions of this bug happening, if that's an issue on a few products, or if there's a specific image that consistently results in this.